packages/components/src/components/input-field/input-field.tsx

```diff
@@ -122,7 +122,7 @@ const InputField = ({
 }: TInputField) => {
     const [local_value, setLocalValue] = React.useState<string>();
     const Icon = icon as React.ElementType;
-    const has_error = error_messages && !!error_messages.length;
+    const has_error = error_messages && !!error_messages.length && !is_error_tooltip_hidden;
```

Comment:
```suggestion
    const has_error = error_messages?.length > 0 && !is_error_tooltip_hidden;
```

Reply:
Hi @markwylde-deriv !
As `error_messages` might be undefined, TS won't let us compare `error_messages?.length` with zero
<img width="606" alt="Screenshot 2023-09-06 at 5 32 59 PM" src="https://github.com/binary-com/deriv-app/assets/121025168/23acb87a-0518-4540-8e50-191a851968ad">

Reply:
In that case we will need to write smth like `const has_error = error_messages?.length && error_messages?.length > 0 && !is_error_tooltip_hidden;` but it's almost the same code with which we started)
Please, correct me if I'm wrong as I'm not so experienced

Reply:
Yeah I think you're right :)

Reply:
Yeah I think you're right 😄  I wonder, is error_messages can be undefined, and can also have an empty array, should it just be one or the other.

Would be so nice to do :
```
!error_messages && !is_error_tooltip_hidden
```

But anyway. Was just a passing comment. Ignore this 😄